### PR TITLE
Theme Options: Drop options prop from connectOptions()

### PR DESCRIPTION
**Pull request by ockham**
And compute theme options entirely inside `theme-options.js`, based on `hideForTheme` attributes.

@folletto This should also align the order of theme options to be the same everywhere -- controlled by [this list](https://github.com/Automattic/wp-calypso/blob/109b74ec94fc7de2b575e0da7cf6a8ae9fa96961/client/my-sites/themes/theme-options.js#L156-L168), and the filtering that removes the options that don't apply.

To test -- same as #12383:

> Compare this branch to production. Compare all different showcase modes (logged-out, multi-site, single-site WP.com, single-site Jetpack), different themes (free, premium, purchased premium, active), and different plans (free, premium, business). Make sure that the '...' contains the same entries on this branch as in production. In multi-site mode, also compare the contents of the site selector modal.

### CODE REVIEW

File: client/my-sites/theme/main.jsx [hunk 1]
```diff
@@ -613,15 +613,6 @@ const ThemeSheetWithOptions = ( props ) => {
 	return (
 		<ConnectedThemeSheet { ...props }
 			siteId={ siteId }
-			theme={ props /* TODO: Have connectOptions() only use theme ID */ }
```
Comment (ockham) [inline client/my-sites/theme/main.jsx]: `connectOptions()` hasn't accepted a theme prop for a looong time.

**Comment by folletto** [comment]
1. Logged Out → Free → ••• → check list — 
1. Logged Out → Premium → ••• → check list — 
2. All Sites → ••• → check list — 
3. .com site → ••• → check list — 
4 Jetpack site → ••• → check list — 






Compare all different showcase modes (logged-out, multi-site, single-site WP.com, single-site Jetpack), different themes (free, premium, purchased premium, active), and different plans (free, premium, business). Make sure that the '...' contains the same entries on this branch as in production. In multi-site mode, also compare the contents of the site selector modal.

**Comment by folletto** [comment]
Checked on Chrome:

1. Logged Out → Free → ••• → check list — ✅ 
2. Logged Out → Premium → ••• → check list — ✅ 
3. All Sites → Free → ••• → check list — ✅ 
4. All Sites → Premium → ••• → check list — ✅ 
5. All Sites → Site selector modal — ✅ 
6. Site .com → Free → ••• → check list — ✅ 
7. Site .com → Premium → ••• → check list — ✅ 
8. Site .com → Purchased → ••• → check list — ✅ 
9. Site .com → Active → ••• → check list — ✅ 
10. Jetpack .com → Free → ••• → check list — ❌  
11. Jetpack .com → Uploaded → ••• → check list — ⚠️ 
12. Jetpack .com → Active → ••• → check list — ✅ 

---

**10. Jetpack .com → Free → ••• → check list**

Delete is showing up for ".com" themes.

**11. Jetpack .com → Uploaded → ••• → check list**

"Delete" is not at the bottom of the top lis (should be after "Try & Customize", before the separator.

**Comment by ockham** [comment]
> 10. Jetpack .com → Free → ••• → check list
> Delete is showing up for ".com" themes.

Uh, good spot. This will require a distinguishing feature for the lower list -- I'm thinking of introducing that `origin` prop to `connectOptions` I've been talking about. It'll come in handy for some other things, too.

> 11. Jetpack .com → Uploaded → ••• → check list
> "Delete" is not at the bottom of the top lis (should be after "Try & Customize", before the separator.

Should be fixed by 6aec3d1.

**Comment by ockham** [comment]
> > 10. Jetpack .com → Free → ••• → check list
> > Delete is showing up for ".com" themes.
>
> Uh, good spot. This will require a distinguishing feature for the lower list -- I'm thinking of introducing that `origin` prop to `connectOptions` I've been talking about. It'll come in handy for some other things, too.

Should be fixed by 71fc70f.

**Comment by folletto** [comment]
Checked on Chrome:

1. Logged Out → Free → ••• → check list — ✅ 
2. Logged Out → Premium → ••• → check list — ✅ 
3. All Sites → Free → ••• → check list — ✅ 
4. All Sites → Premium → ••• → check list — ✅ 
5. All Sites → Site selector modal — ✅ 
6. Site .com → Free → ••• → check list — ✅ 
7. Site .com → Premium → ••• → check list — ✅ 
8. Site .com → Purchased → ••• → check list — ✅ 
9. Site .com → Active → ••• → check list — ✅ 
10. Jetpack .com → Free → ••• → check list — ✅ 
11. Jetpack .com → Uploaded → ••• → check list — ✅ 
12. Jetpack .com → Active → ••• → check list — ✅

:shipit:

**Pull request merged**